src/EventListener/DocumentUploadSubscriber.php

```diff
@@ -2,48 +2,50 @@
 
 namespace GemeenteAmsterdam\FixxxSchuldhulp\EventListener;
 
-use GemeenteAmsterdam\FixxxSchuldhulp\Azure\AzureStorage;
-use Doctrine\Common\EventSubscriber;
-use Doctrine\ORM\Event\LifecycleEventArgs;
+use Doctrine\Bundle\DoctrineBundle\Attribute\AsDoctrineListener;
+use Doctrine\Bundle\DoctrineBundle\EventSubscriber\EventSubscriberInterface;
+use Doctrine\ORM\Event\PostRemoveEventArgs;
+use Doctrine\ORM\Event\PrePersistEventArgs;
+use Doctrine\ORM\Events;
+
 use GemeenteAmsterdam\FixxxSchuldhulp\Entity\Document;
+use GemeenteAmsterdam\FixxxSchuldhulp\Azure\AzureStorage;
 use GemeenteAmsterdam\FixxxSchuldhulp\Service\FileStorageSelector;
+
 use Psr\Log\LoggerInterface;
 use Symfony\Component\HttpFoundation\File\UploadedFile;
 
-class DocumentUploadSubscriber implements EventSubscriber
+#[AsDoctrineListener(event: Events::prePersist, priority: 500, connection: 'default')]
+#[AsDoctrineListener(event: Events::postRemove, priority: 500, connection: 'default')]
+class DocumentUploadSubscriber implements EventSubscriberInterface
 {
-
     /**
      * @param FileStorageSelector $fileStorageSelector
      * @param LoggerInterface $logger
-     * @param AzureStorage $azureStorage
      */
     public function __construct(
         protected FileStorageSelector $fileStorageSelector,
         protected LoggerInterface     $logger,
-        protected AzureStorage        $azureStorage
```

Comment:
Kunnen we niet net zo goed alle code die met AzureStorage te maken heeft ook direct verwijderen?

Reply:
Is gedaan :-)